### PR TITLE
swap value functions for Round sprite hack.

### PR DESCRIPTION
**Pull request by ssakash**

### CODE REVIEW

**Comment by ssakash** [comment]
should have done this ages ago :p , many people requested for this change. @karasuhebi could explain. :)

since, the checked box means something of full value and squared box means something which is around half enabled, which now is sync with the variable's usage in user's perspective.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 1]
```diff
@@ -565,7 +565,20 @@ GSHacksDlg::GSHacksDlg() :
 	memset(msaa2cb, 0, sizeof(msaa2cb));
 	memset(cb2msaa, 0, sizeof(cb2msaa));
 }
-
+int swap_states(int a)
+{
+	switch(a)
+	{
+		case 0: return 0;
+		break;
```
Comment (gregory38) [inline plugins/GSdx/GSSettingsDlg.cpp]: Don't do break after return

Comment (ssakash) [inline plugins/GSdx/GSSettingsDlg.cpp]: sorry, one of my old habit's. fixed :)

**Comment by karasuhebi** [comment]
Yay!

**Comment by turtleli** [comment]
Would a dropdown box be more effective instead? It also helps with message reuse on Linux.
The final part of the round sprite hack help message makes no sense for the dropdown box used in Linux:
```
"Carries a total of three states: Unchecked (off), Shaded (on for all sprites) and Checked (on for flat sprites)"
```
The same applies for the Sprite hack too, since Half = Checked, Full = Shaded.

**Comment by karasuhebi** [comment]
I would greatly prefer a combobox, yes. Not just with this hack, with the other ones that have more than two possible states.

**Comment by ssakash** [comment]
well, there is a reason on why three state checkboxes are present. I think combo box would ideal for setting's with more than three states. also, the three state checkbox have been used from the 1.2.1 release so I think it would be better if we make the change after getting tons of opinions from the user's on the Idea of combo box. for now, the PR is finalized to set up the initial behavior of the three state checkboxes.

**Comment by ssakash** [comment]
@refractionpcsx2  https://www.dropbox.com/s/men4u3pjz1oc5wg/patch-22.7z?dl=0

**Comment by tsunami2311** [comment]
Might want to double check everything is right, Round sprite=Box use to fix( it removed it) the line in middle of Xenosaga cut scenes, with this change the fix should now be Round sprite=check.

But that no longer removes the line it just make it less noticeable but it is still there

**Comment by ssakash** [comment]
Hmm, there was no internal function change in this commit. Only the swapping of values taken by the variable, the problem should have been present some where before then.

**Comment by tsunami2311** [comment]
well i  went from build "28815db" to "621d1d9" and since then it not long remove the line, so either something in the code change or something inthe gdsx changes cause it, cause atm it no long works  for removing the line it just causing  TF to only work on  3d

**Pull request merged**